File: Example/JFMinimalNotification/DetailViewController.m
//
//  DetailViewController.m
//  JFMinimalNotification
//
//  Created by Jeremy Fox on 5/1/15.
//  Copyright (c) 2015 Jeremy Fox. All rights reserved.
//

#import "DetailViewController.h"
#import "JFMinimalNotification.h"

@interface DetailViewController ()
@property (nonatomic, strong) JFMinimalNotification* notification;
@end

@implementation DetailViewController

- (void)dealloc
{
    self.notification = nil;
}

- (void)viewWillAppear:(BOOL)animated {
    [super viewWillAppear:animated];
    self.notification = [JFMinimalNotification notificationWithStyle:JFMinimalNotificationStyleSuccess title:@"Testing" subTitle:@"Testing 1 2 3" dismissalDelay:2.0];
    [self.view addSubview:self.notification];
    self.notification.presentFromTop = YES;
}

- (void)viewDidAppear:(BOOL)animated {
    [super viewDidAppear:animated];
    [self.notification show];
}

- (IBAction)dismissAndShowNotificationAfterDelayAction:(id)sender {
    
    // This is to simulate the view being dismissed, and therby deallocated, before the notification is shown. This causes a similar crash as reported here https://github.com/atljeremy/JFMinimalNotifications/issues/10
    [self.navigationController popViewControllerAnimated:YES];
    
}

@end

File: Example/JFMinimalNotification/DetailViewController.swift
//
//  DetailViewController.swift
//  JFMinimalNotification
//
//  Created by Jeremy E Fox on 9/26/24.
//  Copyright © 2024 Jeremy Fox. All rights reserved.
//

import UIKit

class DetailViewController: UIViewController {
    
    var notification: JFMinimalNotification?
    
    deinit {
        notification = nil
    }
    
    override func viewWillAppear(_ animated: Bool) {
        super.viewWillAppear(animated)
        
        notification = JFMinimalNotification(
            withStyle: .success,
            title: "Testing",
            subTitle: "Testing 1 2 3",
            dismissalDelay: 2.0,
            touchHandler: { [weak self] in
                self?.notification?.dismiss()
            }
        )
        notification?.edgePadding = UIEdgeInsets(top: 10, left: 0, bottom: 0, right: 0)
        
        if let notificationView = notification {
            navigationController?.view.addSubview(notificationView)
        }
        
        notification?.presentFromTop = true
    }
    
    override func viewDidAppear(_ animated: Bool) {
        super.viewDidAppear(animated)
        notification?.show()
    }
    
    @IBAction func dismissAndShowNotificationAfterDelayAction(_ sender: Any) {
        // This simulates the view being dismissed and deallocated before the notification is shown.
        // This is to reproduce a potential crash similar to https://github.com/atljeremy/JFMinimalNotifications/issues/10
        navigationController?.popViewController(animated: true)
    }
}
